### ISSUE TITLE
Particle Seeding Bug

### ISSUE THREAD

**Issue by PavelBlend** [rlguy/Blender-FLIP-Fluids#509]
#### System Information

**Blender Version (including hash):** 2.90.1 3e85bb34d0d7
**FLIP Fluids Version:** 1.0.9
**Operating System:** Windows 10 64 bit
**CPU:** AMD A8 5600K
**GFX:** -
**RAM:** 16 GB

#### Describe the bug

I'm not sure if this is a bug or not. I have done a liquid test with a huge number of substeps (64). I noticed that particles are born too often in some local areas. Perhaps this error is visible even with a lower number of substeps, although this error will not be so noticeable.

#### To Reproduce

Bake liquid from this file:
[particles_seeding_bug.zip](https://github.com/rlguy/Blender-FLIP-Fluids/files/5683631/particles_seeding_bug.zip)

#### Expected Behaviour

Particles should not be born at such a close distance from each other.

#### Actual Behaviour

The particles are too close to each other, they are lined up.

#### Screenshots

![particle_bug](https://user-images.githubusercontent.com/7983249/101993045-65901800-3cc8-11eb-8074-b96a9d6e7d20.jpg)
![seeding_2](https://user-images.githubusercontent.com/7983249/101993064-9708e380-3cc8-11eb-836d-134f26e8e3ee.jpg)

**Comment by rlguy** [comment]
Hi PavelBlend,

Thanks for the report! It looks like this issue is caused by unexpected behavior of the random number generator (RNG) that we are using. The RNG is generating the same 'random' positions for every substep which is creating long lines of particles in some locations.

I have learned that the current RNG technique we are using (C++ _rand()_) is not good and I have updated the RNG method to a better technique ([_Mersenne Twister_](https://stackoverflow.com/a/26853142)), which should fix this issue:

![new_rng](https://user-images.githubusercontent.com/4285191/102149188-b4ea5b80-3e22-11eb-9071-f9b769fdd662.jpeg)

I will leave this issue open for now. I still need to test this on MacOS and Linux to make sure it works. I'll update later whent his fix will be available.
